### PR TITLE
Add section on reporting bugs to contributing guide

### PR DESCRIPTION
**Pull request by izuzak**
This adds a section on reporting bugs to the contributing guide and removes the generic section on submitting issues. My plan is to add a section on submitting feature requests and sections on other types of "issues" and pull requests, so that we cover all the different types of contributions. I think that will be more helpful than a single generic section since it will allow us to provide specific advice for each type of contribution, and I also think users will find the right section easier.

The section on submitting bug reports is structured in three parts: what you should do before submitting a bug report, which information you should provide in a bug report, and a template for structuring the information. For the part on which information should be provided, I tried to structure this as a list, and bold key sections off each note so that it's easy to scan. I think it will be helpful to list all the questions we often ask when debugging issues -- sometimes that will help users figure out things on their own and provide more helpful reports, and it will allow maintainers to link to this section and say "Please provide more information, as listed here". 

@atom/feedback @atom/issue-triage As always, I'd appreciate your feedback. :honey_pot: Is there any other information we normally ask from users when debugging issues? How do you like the idea of providing a template here, and should some other questions/information be a part of it?

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -71,30 +71,91 @@ For more information on how to work with Atom's official packages, see [Contribu
...
+* **Describe the behavior you observed after following the steps** and point out what exactly is the problem with that behavior.
+* **Explain which behavior you expected to see instead and why.**
+* **Include screenshots and animated GIFs** which show you following the described steps and clearly demonstrate the problem. If you use the keyboard while following the steps, **record the GIF with the [Keybinding Resolver](https://github.com/atom/keybinding-resolver) shown**. You can use [this tool](http://www.cockos.com/licecap/) to record GIFs on OSX and Windows.
+* **If you're reporting that Atom crashed**, include a crash report with a stack trace from the operating system. On OSX, the crash report will be available in `Console.app` under "Diagnostic and usage information" > "User diagnostic reports". Include the crash report in the issue in a [code block](https://help.github.com/articles/markdown-basics/#multiple-lines) or put it in a [gist](https://gist.github.com/) and provide link to that gist.
```
Comment (izuzak) [inline CONTRIBUTING.md]: How does one obtain crash reports on Windows and Linux? I have no idea, but would love to add links :link: or instructions :memo: here.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -71,30 +71,91 @@ For more information on how to work with Atom's official packages, see [Contribu
...
+* **Provide specific examples to demonstrate the steps**. Include links to files or GitHub projects, or copy/pasteable snippets, which you use in those examples. If you're providing snippets in the issue, use [Markdown code blocks](https://help.github.com/articles/markdown-basics/#multiple-lines).
+* **Describe the behavior you observed after following the steps** and point out what exactly is the problem with that behavior.
+* **Explain which behavior you expected to see instead and why.**
+* **Include screenshots and animated GIFs** which show you following the described steps and clearly demonstrate the problem. If you use the keyboard while following the steps, **record the GIF with the [Keybinding Resolver](https://github.com/atom/keybinding-resolver) shown**. You can use [this tool](http://www.cockos.com/licecap/) to record GIFs on OSX and Windows.
```
Comment (izuzak) [inline CONTRIBUTING.md]: Is there an easy-to-use tool (similar to licecap) for recording GIFs on Linux?

**Comment by izuzak** [comment]
I'll probably merge this tomorrow, so if anyone from @atom/feedback or @atom/issue-triage has any 💭 or 💡 about this, or knows the answers to [these](https://github.com/atom/atom/pull/9333/files#r43386092) [two](https://github.com/atom/atom/pull/9333/files#r43386290) questions -- please let me know. 🙏

File: CONTRIBUTING.md (hunk 2 above)
Comment (mnquintana) [inline CONTRIBUTING.md]: I think I used [byzanz](https://github.com/GNOME/byzanz) on Ubuntu, but it's nowhere near as easy to use as LICEcap. Most Linux solutions for this are rather janky Bash scripts that require you to manually figure out the exact pixel position of the window you want to capture.

Comment (izuzak) [inline CONTRIBUTING.md]: >  think I used byzanz on Ubuntu, but it's nowhere near as easy to use as LICEcap. 

Thanks for that link, @mnquintana :bow:. I added that and another tool to the list. Hopefully, that will help at least a bit.

**Comment by izuzak** [comment]
If anyone has any suggestions about improving this section of the contributing guide -- please open an issue or pull request. :open_hands:

**Pull request merged**